Intermittent CI failure in step `Install Carrier`. Deployment troubles, timeout in waiting for completion.

```
Deploying Quarks
⣾ Waiting for resource  to be running in quarks ... 💤   ...
Couldn't install carrier: failed waiting Quarks deployment to come up: no pods in quarks with selector 
Error: Process completed with exit code 1.
```

--- a/cli/deployments/eirini.go
+++ b/cli/deployments/eirini.go
@@ -169,17 +169,16 @@ func (k Eirini) apply(c kubernetes.Cluster, options kubernetes.InstallationOptio
 		return err
 	}
 
-	if err := c.WaitUntilPodBySelectorExist("eirini-core", "name=eirini-api", k.Timeout); err != nil {
-		return errors.Wrap(err, "failed waiting Eirini api deployment to exist")
-	}
-	if err := c.WaitForPodBySelectorRunning("eirini-core", "name=eirini-api", k.Timeout); err != nil {
-		return errors.Wrap(err, "failed waiting Eirini api deployment to come up")
-	}
-	if err := c.WaitUntilPodBySelectorExist("eirini-core", "name=eirini-metrics", k.Timeout); err != nil {
-		return errors.Wrap(err, "failed waiting Eirini metrics deployment to exist")
-	}
-	if err := c.WaitForPodBySelectorRunning("eirini-core", "name=eirini-metrics", k.Timeout); err != nil {
-		return errors.Wrap(err, "failed waiting Eirini metrics deployment to come up")
+	for _, podname := range []string{
+		"eirini-api",
+		"eirini-metrics",
+	} {
+		if err := c.WaitUntilPodBySelectorExist("eirini-core", "name="+podname, k.Timeout); err != nil {
+			return errors.Wrap(err, "failed waiting Eirini "+podname+" deployment to exist")
+		}
+		if err := c.WaitForPodBySelectorRunning("eirini-core", "name="+podname, k.Timeout); err != nil {
+			return errors.Wrap(err, "failed waiting Eirini "+podname+" deployment to come up")
+		}
 	}
 
 	emoji.Println(":heavy_check_mark: Eirini deployed")

--- a/cli/deployments/gitea.go
+++ b/cli/deployments/gitea.go
@@ -134,8 +134,17 @@ gitea:
 		return errors.New("Failed installing Gitea: " + out)
 	}
 
-	if err := c.WaitForPodBySelectorRunning(giteaDeploymentID, "", k.Timeout); err != nil {
-		return errors.Wrap(err, "failed waiting Gitea deployment to come up")
+	for _, podname := range []string{
+		"memcached",
+		"postgresql",
+		"gitea",
+	} {
+		if err := c.WaitUntilPodBySelectorExist(giteaDeploymentID, "app.kubernetes.io/name="+podname, k.Timeout); err != nil {
+			return errors.Wrap(err, "failed waiting Gitea "+podname+" deployment to exist")
+		}
+		if err := c.WaitForPodBySelectorRunning(giteaDeploymentID, "app.kubernetes.io/name="+podname, k.Timeout); err != nil {
+			return errors.Wrap(err, "failed waiting Gitea "+podname+" deployment to come up")
+		}
 	}
 
 	emoji.Println(":heavy_check_mark: Gitea deployed")

--- a/cli/deployments/quarks.go
+++ b/cli/deployments/quarks.go
@@ -67,8 +67,17 @@ func (k Quarks) apply(c kubernetes.Cluster, options kubernetes.InstallationOptio
 		return errors.New("Failed installing Quarks")
 	}
 
-	if err := c.WaitForPodBySelectorRunning(quarksDeploymentID, "", k.Timeout); err != nil {
-		return errors.Wrap(err, "failed waiting Quarks deployment to come up")
+	for _, podname := range []string{
+		"cf-operator",
+		"quarks-secret",
+		"quarks-job",
+	} {
+		if err := c.WaitUntilPodBySelectorExist(quarksDeploymentID, "name="+podname, k.Timeout); err != nil {
+			return errors.Wrap(err, "failed waiting Quarks "+podname+" deployment to exist")
+		}
+		if err := c.WaitForPodBySelectorRunning(quarksDeploymentID, "name="+podname, k.Timeout); err != nil {
+			return errors.Wrap(err, "failed waiting Quarks "+podname+" deployment to come up")
+		}
 	}
 
 	emoji.Println(":heavy_check_mark: Quarks deployed")

--- a/cli/deployments/traefik.go
+++ b/cli/deployments/traefik.go
@@ -70,7 +70,10 @@ func (k Traefik) apply(c kubernetes.Cluster, options kubernetes.InstallationOpti
 		return errors.Wrap(err, fmt.Sprintf("Failed installing Traefik: %s\n", out))
 	}
 
-	if err := c.WaitForPodBySelectorRunning(traefikDeploymentID, "", k.Timeout); err != nil {
+	if err := c.WaitUntilPodBySelectorExist(traefikDeploymentID, "app.kubernetes.io/name=traefik", k.Timeout); err != nil {
+		return errors.Wrap(err, "failed waiting Traefik Ingress deployment to exist")
+	}
+	if err := c.WaitForPodBySelectorRunning(traefikDeploymentID, "app.kubernetes.io/name=traefik", k.Timeout); err != nil {
 		return errors.Wrap(err, "failed waiting Traefik Ingress deployment to come up")
 	}
 

